### DISCUSSION TITLE
Valibot Code Generation

### DISCUSSION THREAD

**Discussion by sinclairzx81** [open-circle/valibot#57] [Show and tell]
Hi, [TypeBox](https://github.com/sinclairzx81/typebox) developer here. I came across Valibot the other day, and the library design looks great! (also I like the robot mascot)

Just posting as I've started writing some tooling that converts TypeScript type definitions into Valibot types (as well as a several other ecosystem libraries) and thought I'd share in case it's of some use to the Valibot project. The tooling was originally written for TypeBox to help get its types lined up to TypeScript, but have since extended support for other libraries (as it's quite helpful to see TS types and library types side-by-side), especially when trying replicate various parts of the type system.

These tools are two part, a code editor website and a code generation library, you can find both at the links below.

### Website

https://sinclairzx81.github.io/typebox-workbench/

### Code Generation

https://github.com/sinclairzx81/typebox-codegen

If these are of some use, and you'd like me to keep Valibot codegen current with new types / features, feel free to ping me on either of these projects (ideally the workbench as prototyping usually happens there first before being moved to the codegen for npm publishing). I'll be trying to keep these libraries as current and accurate as I can (mostly as a side project, and more so if the tools are found to be useful to the ecosystem), however a ping can help get newer library functionality up and integrated quicker. PR's are also welcome.

Again, Good work on Valibot!
Cheers
S

**Comment by fabian-hiller** [comment]
Wow! Feel free to add the project to our [ecosystem page](https://valibot.dev/guides/ecosystem/). There have been multiple requests for JSON to Valibot conversion (and Valibot to JSON as well). Feel free to keep it updated. Thank you! I think Valibot has the chance to become a bigger project in the coming years.

- https://github.com/fabian-hiller/valibot/issues/23
- https://github.com/fabian-hiller/valibot/issues/30

**Comment by sinclairzx81** [comment]
Code generation updated for Valibot 0.13.1. Can test via https://sinclairzx81.github.io/typebox-workbench/

Added:
- nullType
- undefinedType
- integer (number via integer contraint)
- voidType

**Reply by fabian-hiller** [reply]
Thank you very much! I'll try to remember to let you know if something changes with an update.
